prefect_utils/mysql.py

```diff
@@ -0,0 +1,103 @@
+"""
+Tasks for interacting with Aurora MySQL.
+"""
+import mysql.connector
+from prefect import task
+from prefect.engine import signals
+from prefect.utilities.logging import get_logger
+
+
+@task
+def load_s3_data_to_mysql(
+    aurora_credentials: dict,
+    database: str,
+    table: str,
+    table_schema: list,
+    s3_url: str,
+    field_delimiter: str = ',',
+    enclosed_by: str = '',
+    escaped_by: str = '\\\\',
+    overwrite: bool = False,
+):
+
+    """
+    Loads an Aurora MySQL database table from text files in S3.
+
+    Args:
+      aurora_credentials (dict): Aurora credentials dict containing `user`, `password` and `host`.
+      database (str): Name of the destination database.
+      table (str): Name of the destination table.
+      table_schema (list): List of tuples specifying table schema.
+              Example: `[('id', 'int'), ('course_id', 'varchar(255) NOT NULL')]`
+      s3_url (str): Full S3 URL containing the files to load into the destination table .
+      field_delimiter (str, optional): The character used to indicate how the fields in input files are delimited.
+              Defaults to `,`.
+      enclosed_by (str, optional): Single character string that specifies the fields enclosing character.
+              Defaults to empty string(no enclosing character).
+      escaped_by (str, optional): Single character string which indicates the escaping of delimiters and
+              other escape sequences. Defaults to backslack(`\\`),
+      overwrite (bool, optional): Whether to overwrite existing data in the destination Table. Defaults to `False`.
+    """
+
+    logger = get_logger()
+
+    user = aurora_credentials.get('user')
+    password = aurora_credentials.get('password')
+    host = aurora_credentials.get('host')
+
+    connection = mysql.connector.connect(
+        user=user,
+        password=password,
+        host=host,
+        database=database
+    )
+
+    cursor = connection.cursor()
+
+    table_schema = ','.join(
+        '{name} {definition}'.format(name=name, definition=definition) for name, definition in table_schema
+    )
+
+    query = """
+        CREATE TABLE IF NOT EXISTS {table} ({table_schema})
```

Comment:
It doesn't seem like there is a way to drop and recreate the table if the schema changes. It complicates things a little, but maybe we should have an option that makes this a `CREATE OR REPLACE`?

Reply:
For full replacement, we can do the loading in another newly created table with new schema, and do an atomic rename. I don't see `CREATE OR REPLACE` for mysql, even if we could, that would cause an implicit commit and there would not be any data for the duration of `LOAD DATA FROM S3` command.

Not sure how to handle this for incremental loads. Do you have any thoughts on that ?

Reply:
Is this too simplistic?

On non-overwrite: do a query that checks for existence of rows that meet the `record_filter` and skip if any exist.

On overwrite: run a DELETE with the `record_filter` then load.

Reply:
Wait you already do that... did that change just go in?

Reply:
Yes, I think I pushed those changes around the time you typed that reply.

Reply:
I was referring to making schema changes work with incrementally loaded tables. I'm not so sure if there is a straightforward way of doing that without increasing the complexity of the code. 
For full refresh loads, schema changes can be made by making use of a temp table followed by rename. But that would not work with incremental tables as historical data would be lost unless we do some sort of copy magic:
create new table with new schema -> insert existing data from primary table -> rename.

Reply:
Makes sense 👍🏼